src/array.rs

```diff
@@ -50,6 +50,34 @@ where
         self.push_unchecked(src);
     }
 
+    /// Remove the element at pos and return it
+    pub fn remove(&mut self, pos: usize) -> u8 {
+        let val = self[pos];
+        self.as_mut_bytes().split_at_mut(pos).1.rotate_left(1);
+        self.length -= 1;
+        val
+    }
+
+    ///Insert the element at pos without validating
+    ///
+    /// Panics if the capacity is full or if the pos is > self.len()
+    pub fn insert_unchecked(&mut self, pos: usize, byte: u8) {
+        let curr_len = self.len();
+        assert!(
+            curr_len <= (N - 1),
+            "No remaining capacity to insert a byte into the ByteArray",
+        );
+        assert!(
+            pos <= curr_len,
+            "Cannot insert at position {} that is more that the array length {}.",
+            pos,
+            curr_len
```

Comment:
Rust finally has (at least simple) inline format args - it works for values, but sadly not expressions yet.
```suggestion
            "Cannot insert at position {pos} that is more that the array length {curr_len}.",
```

Reply:
Done! Had to update the rust edition to 2021 to get this feature :)